components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -17,9 +20,27 @@ ZCashKeyring::ZCashKeyring(base::span<const uint8_t> seed, bool testnet)
           seed,
           GetRootPath(testnet ? mojom::KeyringId::kZCashTestnet
                               : mojom::KeyringId::kZCashMainnet)),
-      testnet_(testnet) {}
+      testnet_(testnet) {
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (!seed.empty() && IsZCashShieldedTransactionsEnabled()) {
```

Comment:
We just silently fail to do anything if the seed is empty and shielded transactions are enabled? Doesn't that leave the keyring in a bad state that is going to cause failures later?

Reply:
Same behavior is used in the parent class Secp256k1HDKeyring, if seed is empty then root hd key is empty and we can't just use keyring properly.

Reply:
They can be handled in a static function like `CreateZCashKeyring`, if it doesn't get satisfied, the keyring should not be created and kept in KeyringService.

Reply:
yes, you could make the constructor private and do something like that to ensure that it's valid first before using it

Reply:
these are fine for follow-up I guess if other keyrings do this already

Reply:
https://github.com/brave/brave-browser/issues/37858